tests/components/vesync/test_init.py

```diff
@@ -36,6 +40,40 @@ async def test_async_setup_entry__not_login(
     assert not hass.data.get(DOMAIN)
 
 
+async def test_async_setup_entry_VeSyncAPIResponseError(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    manager: VeSync,
+) -> None:
+    """Test setup does not create config entry when not logged in."""
+    manager.login = AsyncMock(side_effect=VeSyncAPIResponseError("Mock login failed"))
+
+    assert not await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert manager.login.call_count == 1
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert config_entry.state is ConfigEntryState.SETUP_RETRY
+    assert not hass.data.get(DOMAIN)
+
+
+async def test_async_setup_entry_VeSyncServerError(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    manager: VeSync,
+) -> None:
+    """Test setup does not create config entry when not logged in."""
+    manager.login = AsyncMock(side_effect=VeSyncServerError("Mock login failed"))
+
+    assert not await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert manager.login.call_count == 1
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert config_entry.state is ConfigEntryState.SETUP_RETRY
+    assert not hass.data.get(DOMAIN)
```

Comment:
1. we should use snake_case for the test name
2. we can parametrize the tests
3. `assert not hass.data.get(DOMAIN)` isn't needed as ideally we're going to store it in the entry.runtime_data